### PR TITLE
Google OAuth and JWT token 

### PR DESCRIPTION
**Pull request by ThomasGassmann**

### CODE REVIEW

**Comment by alexisrolland** [comment]
@ThomasGassmann could you please create the PR on the development branch rather than master? :)
Thanks

**Comment by ThomasGassmann** [comment]
> @ThomasGassmann could you please create the PR on the development branch rather than master? :)
> Thanks

Retargeted

File: api/init/security/decorators.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def token_required(func):
+    """Gets a decorator to check whether a valid token is in the request headers. It will return
+        401, if the user provdies an invalid token.
+    """
+    def wrapper(*args, **kwargs):
+        auth_header = _get_token_from_header()
+        token = _parse_jwt_token(auth_header)
+        unauthorized_response = make_response(jsonify({}), 401)
```
Comment (PascalHonegger) [inline api/init/security/decorators.py]: Add a message to the return body

File: app/src/Components/Login/Login.js [hunk 2]
```diff
@@ -1,35 +1,34 @@
 import React from 'react';
```
Comment (PascalHonegger) [inline app/src/Components/Login/Login.js]: Revert formatting changes to prevent future merge conflicts

Comment (ThomasGassmann) [reply app/src/Components/Login/Login.js]: See 0aafaafba016b3ecb6d62307026422341f1e620a

File: api/init/security/decorators.py (hunk 1 above)
Comment (ThomasGassmann) [reply api/init/security/decorators.py]: See 0aafaafba016b3ecb6d62307026422341f1e620a

File: api/init/proxy/utils.py [hunk 3]
```diff
@@ -17,6 +17,8 @@ def execute_graphql_request(payload: dict):
 
     url = 'http://graphql:5433/graphql'  # Should be moved to config file
     headers = {'Content-Type': 'application/graphql'}
+
+    print(payload['query'])
```
[the remark this answers is not in the document]
Comment (ThomasGassmann) [reply api/init/proxy/utils.py]: See 0aafaafba016b3ecb6d62307026422341f1e620a

**Pull request merged**